core/templates/pages/collection-editor-page/navbar/collection-editor-navbar.directive.ts

```diff
@@ -205,11 +205,6 @@ angular.module('oppia').directive('collectionEditorNavbar', [
             RouterService.navigateToHistoryTab();
           };
           ctrl.$onInit = function() {
-            ctrl.directiveSubscriptions.add(
-              CollectionEditorStateService.onCollectionInitialized.subscribe(
-                () => _validateCollection()
-              )
-            );
```

Comment:
@kevintab95 Is this okay to remove this?

Reply:
Also, can you help @arpit1912 by telling the purpose of this validation?

Reply:
Nice catch @nishantwrp!

I don't think we should remove this method because it validates the collection and prevents saving of bad collections. @arpit1912 can you please explain why this has been removed? Thanks!

Reply:
ohhk, so @kevintab95 this function validate a collection based on there ```CollectionRights```, now as we are not intialising the ```CollectionRights``` at the time of creation, the ```validateCollection``` function will give errors as it trying to check the ```isprivate``` property of a undefined variable.

Reply:
When exactly is `ctrl.collectionRights` set? Would it be possible to execute the validation after that is set?

Reply:
On line 216 we are setting the ```ctrl.collectionRights``` as follows
``` ctrl.collectionRights = (CollectionEditorStateService.getCollectionRights());```